### ISSUE TITLE
Holey arrays appearing incorrect in console.log

### ISSUE THREAD

**Issue by mostik** [oven-sh/bun#7567]
### What version of Bun is running?

1.0.16+800fb1290

### What platform is your computer?

Linux 6.1.64-1-MANJARO x86_64

### What steps can reproduce the bug?

```js
let a = [1, , ,2]

console.log(a)
```

### What is the expected behavior?

```js
[ 1, 2 x empty items, 2 ]
```

### What do you see instead?

```js
[ 12 x empty items, 2 ]
```

**Comment by paperclover** [comment]
~~Fixed in 1.0.16~~ nevermind

**Comment by kaioduarte** [comment]
Seems fixed on 1.0.20

**Comment by Electroid** [comment]
Closed by #7751
